File: kotlin/ex4_2_2.kt
abstract class Comparable<T> {
    abstract fun gt(that: T): Boolean
}

fun <T : Comparable<T>> sort(lst: List<T>): Unit {
    if (lst[0].gt(lst[1])) {}
}

class Person(val age: Int) : Comparable<Person>() {
    override fun gt(that: Person): Boolean = this.age > that.age
}

fun main() {
    val people: List<Person> = listOf()
    sort<Person>(people)
}

File: scala/ex4_2_2.scala
abstract class Comparable[T]:
  def gt(that: T): Boolean

def sort[T <: Comparable[T]](lst: List[T]): Unit =
  if lst(0).gt(lst(1)) then ()

class Person(val age: Int) extends Comparable[Person]:
  def gt(that: Person): Boolean = this.age > that.age

@main def main =
  val people: List[Person] = List()
  sort[Person](people)
